File: 24-Template-Method/src/go/black_coffee.go
package main

import "fmt"

// BlackCoffee overrides the WantsCondiments hook to skip the condiment step.
type BlackCoffee struct {
	DefaultBeverage
}

func (BlackCoffee) Brew()             { fmt.Println("  Dripping coffee through filter") }
func (BlackCoffee) AddCondiments()    {}
func (BlackCoffee) WantsCondiments() bool { return false }

File: 24-Template-Method/src/rust/black_coffee.rs
use crate::beverage::CaffeineBeverage;

/// Black coffee overrides the hook to skip condiments.
pub struct BlackCoffee;

impl CaffeineBeverage for BlackCoffee {
    fn brew(&self) {
        println!("  Dripping coffee through filter");
    }

    fn add_condiments(&self) {
        // no condiments
    }

    fn wants_condiments(&self) -> bool {
        false // hook overridden — the condiment step is skipped
    }
}
